### DISCUSSION TITLE
Error Setting up EmailJS in NodeJS App - '535 5.7.3 Authentication unsuccessful

### DISCUSSION THREAD

**Discussion by ObjC17** [emailjs-com/help#129] [Q&A]
Hi, 
I wanted to setup EmailJS and i keep getting error as follows :
Error sending email: Error: Invalid login: 535 5.7.3 Authentication unsuccessful [CH0PR04CA0050.namprd04.prod.outlook.com 2024-07-25T06:52:12.382Z 08DCABD98DD48F9D]

   at TLSSocket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:228:10) {
  code: 'EAUTH',
  response: '535 5.7.3 Authentication unsuccessful [CH0PR04CA0050.namprd04.prod.outlook.com 2024-07-25T06:52:12.382Z 08DCABD98DD48F9D]',
  responseCode: 535,
  command: 'AUTH LOGIN'

 in my simple NodeApp. i created .env file and put following values :

- EMAILJS_USER_ID=/* my EmailJS public key
- EMAILJS_SERVICE_ID=/* Service ID from EmailJS portal */
- EMAILJS_TEMPLATE_ID=/* Template ID from EmailJS portal */
- EMAILJS_PASSWORD=/*password */

**and in my server.js file i have following code  :**
----------------------------------------------------------
const express = require("express");
const bodyParser = require("body-parser");
const nodemailer = require("nodemailer");
require("dotenv").config();

const app = express();
const port = process.env.PORT || 3000;

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));
app.use(express.static("public"));

app.post("/send-email", (req, res) => {
  const { email, comments } = req.body;

  // Log request body for debugging
  console.log("Request Body:", req.body);

  // Configure nodemailer

  const transporter = nodemailer.createTransport({
    host: "smtp.office365.com", // Outlook SMTP server
    port: 587, // Port for TLS
    secure: false, // Use SSL/TLS when true for port 465, false for port 587
    auth: {
      user: process.env.EMAILJS_USER_ID, // Your Outlook email address
      pass: process.env.EMAILJS_PASSWORD, // Your Outlook email password or app-specific password
    },
  });

  const mailOptions = {
    from: process.env.EMAILJS_USER_ID,
    to: email,
    subject: "Feedback Received",
    text: comments,
  };

  transporter.sendMail(mailOptions, (error, info) => {
    if (error) {
      console.error("Error sending email:", error);
      return res.status(500).send({ message: "Error sending email", error });
    }
    res.send({ message: "Email sent successfully", info });
  });
});

app.listen(port, () => {
  console.log(`Server is running on port ${port}`);
});

Please help where am i going wrong

**Comment by xr0master** [comment]
You are not using EmailJS, so I am unsure what your question is.

**Reply by ObjC17** [reply]
i am using EmailJS, when i did client side implementation, it was going thru but credentials were exposed so i made my code NodeJS and used Node mailer and shifted the code. Once i did that this error start happening. Do you need any further details?

**Reply by xr0master** [reply]
You use the nodemailer but want us to help you find your issues in the code. Is it correct?
Please go to the nodemailer developer and ask him. Or use the EmailJS service.
https://www.npmjs.com/package/@emailjs/nodejs

**Comment by xr0master** [comment]
Not related to the EmailJS service
